core/src/test/java/io/jenkins/pluginhealth/scoring/probes/SecurityScanGithubWorkflowProbeTest.java

```diff
@@ -0,0 +1,210 @@
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("incorrect-file-name.yml"));
+
+        Files.write(workflowFile, List.of(SecurityScanGithubWorkflowProbe.SEARCH_LINE));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldFailIfJobsIsNotConfiguredInGitHubWorkflow() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml@v2"));
+
+        Files.write(workflowFile, List.of(SecurityScanGithubWorkflowProbe.SEARCH_LINE));
+        when(ctx.getScmRepository()).thenReturn(repo);
+
+        final ProbeResult result = probe.apply(plugin, ctx);
+        assertThat(result.status()).isEqualTo(ResultStatus.FAILURE);
+        assertThat(result.message()).isEqualTo("GitHub workflow security scan is not configured in the plugin");
+    }
+
+    @Test
+    void shouldFailIfUsesIsNotConfigured() throws IOException {
+        final Plugin plugin = mock(Plugin.class);
+        final ProbeContext ctx = mock(ProbeContext.class);
+        final SecurityScanGithubWorkflowProbe probe = getSpy();
+
+        when(plugin.getDetails()).thenReturn(Map.of(
+            SCMLinkValidationProbe.KEY, ProbeResult.success(SCMLinkValidationProbe.KEY, ""),
+            LastCommitDateProbe.KEY, ProbeResult.success(LastCommitDateProbe.KEY, "")
+        ));
+        final Path repo = Files.createTempDirectory("foo");
+        Path workflowPath = Files.createDirectories(repo.resolve(".github/workflows"));
+        final Path workflowFile = Files.createFile(workflowPath.resolve("jenkins-security-scan.yaml@v2"));
+
+        Files.write(workflowFile, List.of(
+            "jobs:",
+            "  security-scan:",
+            "    this-is-not-uses: jenkins-infra/jenkins-security-scan/.github/workflows/jenkins-security-scan.yaml@v2"
```

Comment:
I wouldn't do that as you are creating a file with an incorrect syntax.

```suggestion
            "    uses: not-the-workflow-definition-we-are-lookgin-for@v1"
```